### PR TITLE
Cherrypicks from 1.8

### PR DESCRIPTION
**Pull request by SpacemanPaul**
### Reason for this pull request

Cherry picks of #1624, #1625, #1626, #1629, #1630, #1631

### CODE REVIEW

File: datacube/drivers/postgis/_api.py [hunk 1]
```diff
@@ -1525,19 +1525,19 @@ def temporal_extent_by_prod(self, product_id: int) -> tuple[datetime.datetime, d
         query = self.temporal_extent_full().where(Dataset.product_ref == product_id)
         res = self._connection.execute(query)
         for tmin, tmax in res:
-            return (self.time_min.normalise_value(tmin), self.time_max.normalise_value(tmax))
+            return (time_min.normalise_value(tmin), time_max.normalise_value(tmax))
```
Comment (Ariana-B) [inline datacube/drivers/postgis/_api.py:1528]: Wasn't this change the other way around?

File: datacube/index/postgis/_datasets.py [hunk 2]
```diff
@@ -161,19 +161,21 @@ def add(self, dataset: Dataset,
             return dataset
         with self._db_connection(transaction=True) as transaction:
             # 1a. insert (if not already exists)
-            transaction.insert_dataset(dataset.metadata_doc_without_lineage(), dataset.id, dataset.product.id)
```
Comment (Ariana-B) [inline datacube/index/postgis/_datasets.py:164]: The `product_id` logic should be added in but the rest should be kept as it is here, not changed back

**Review by Ariana-B — CHANGES_REQUESTED** [review]

File: datacube/drivers/postgis/_api.py (hunk 1 above)
Comment (SpacemanPaul) [reply datacube/drivers/postgis/_api.py:1528]: Yep, totally muffed that merge.  Sorry, my mind was elsewhere.  This should fix it.

File: datacube/index/postgis/_datasets.py (hunk 2 above)
Comment (SpacemanPaul) [reply datacube/index/postgis/_datasets.py:164]: As above.

**Review by Ariana-B — APPROVED** [review]

**Pull request merged**